examples/step-78/step-78.cc

```diff
@@ -0,0 +1,869 @@
+// The program starts with the usual include files, all of which you should have
+// seen before by now:
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/solution_transfer.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/data_out_stack.h>
+#include <deal.II/base/convergence_table.h>
+
+#include <fstream>
+#include <iostream>
+
+namespace BlackScholesSolver
+{
+  using namespace dealii;
+
+  template <int dim>
+  double coefficient(const Point<dim> &p)
+  {
+    return p.square();
+  }
+
+  template <int dim>
+  Tensor<1, dim> SCoefficient(const Point<dim> &p)
+  {
+    return Tensor<1, dim>(p);
+  }
```

Comment:
Are these functions needed?

Reply:
I think I had these in there with the intention of easily moving to higher dimensions later. I think for the 1D case at lease, we could get rid of these functions and calculate these directly inline where needed.